schema_based_on_diagram.sql

```diff
@@ -0,0 +1,52 @@
+CREATE TABLE patients (
+	id INT GENERATED ALWAYS AS IDENTITY,
+	name VARCHAR(100),
+	date_of_birth DATE,
+	PRIMARY KEY(id)
+);
+
+CREATE TABLE medical_histories (
+	id INT GENERATED ALWAYS AS IDENTITY,
+	admitted_at DATE,
+	status VARCHAR(100),
+	patient_id INT,
+	PRIMARY KEY(id),
+	CONSTRAINT fk_patients FOREIGN KEY(patient_id) REFERENCES patients(id)
+);
+
+CREATE TABLE treatments (
+	id INT GENERATED ALWAYS AS IDENTITY,
+	type VARCHAR(100),
+	name VARCHAR(100),
+	PRIMARY KEY (id)
```

Comment:
<img width="414" alt="image" src="https://github.com/naji-97/Vet-Clinic-dataBases/assets/27709832/450d3322-c4fa-411a-af9e-21bf649f0d15">

The relationship between the `medical_history` and `treatment` tables is many to many. All many-to-many relationships are required to be represented by a join table, `medical_history_has_treatment`

<img width="809" alt="image" src="https://github.com/naji-97/Vet-Clinic-dataBases/assets/27709832/e19bff77-9f33-4a1f-9f47-d9c665002cfd">

```sql
CREATE TABLE medical_histories_has_treatments (
    medical_history_id int refrences medical_histories(id),
    treatment_id int refrences treatments(id),
    );
```

Reply:
Big thanks again.